##mInDel

###Introduction

mInDel allows to find InDel variations between two or more genotypes and to develop InDel markers. mInDel requires next generation sequencing data of genomic DNA. It may be used for measuring InDel variation between populations, for genome wide association studies and for QTL fine mapping.

mInDel mainly implements five modules: the Pre-processing module, the De novo assembly module, the overlap PCR primer design in batch module ,the ePCR mapping module and InDel screening and marker development module. In the following, we describe the mInDel pipeline for InDel marker development. The mInDel pipeline comprises one pre-processing step, three main marker discovery steps and a post-processing step.

It was tested on CentOS6.3 and ubuntu 12.10 server but it should work in most Linux servers.


### - [Installation](https://github.com/lyd0527/mInDel/blob/master/doc/Installation.md)
### - [Quick Start](https://github.com/lyd0527/mInDel/blob/master/doc/Quick_start.md)
### - [Step By Step Tutorial](https://github.com/lyd0527/mInDel/blob/master/doc/Step_by_step_tutorial.md)
### - [Additional](https://github.com/lyd0527/mInDel/blob/master/doc/Additional.md)
*Subroutines may be also run as separate programs for different purposes.*


### QUESTIONS AND COMMENTS:
Please send your any comments to Dr. Yuanda Lv, Lyd0527@jaas.ac.cn.

*Copyright (c) 2014, Institute of Biotechnology, Jiangsu Academy of Agricultural Sciences.*
